examples/step-86/step-86.cc

```diff
@@ -330,26 +341,28 @@ namespace Step86
       DoFTools::extract_locally_relevant_dofs(dof_handler);
 
 
-    hanging_nodes_constraints.clear();
-    hanging_nodes_constraints.reinit(locally_owned_dofs, locally_relevant_dofs);
+    hanging_node_constraints.clear();
+    hanging_node_constraints.reinit(locally_owned_dofs, locally_relevant_dofs);
     DoFTools::make_hanging_node_constraints(dof_handler,
-                                            hanging_nodes_constraints);
-    hanging_nodes_constraints.make_consistent_in_parallel(locally_owned_dofs,
-                                                          locally_relevant_dofs,
-                                                          mpi_communicator);
-    hanging_nodes_constraints.close();
+                                            hanging_node_constraints);
+    hanging_node_constraints.make_consistent_in_parallel(locally_owned_dofs,
+                                                         locally_relevant_dofs,
+                                                         mpi_communicator);
+    hanging_node_constraints.close();
 
 
     homogeneous_constraints.clear();
-    homogeneous_constraints.reinit(locally_owned_dofs, locally_relevant_dofs);
-    homogeneous_constraints.merge(hanging_nodes_constraints);
+    homogeneous_constraints.reinit(locally_owned_dofs,
+                                   hanging_node_constraints.get_local_lines());
+    homogeneous_constraints.merge(hanging_node_constraints);
     VectorTools::interpolate_boundary_values(dof_handler,
                                              0,
                                              Functions::ZeroFunction<dim>(),
                                              homogeneous_constraints);
-    homogeneous_constraints.make_consistent_in_parallel(locally_owned_dofs,
-                                                        locally_relevant_dofs,
-                                                        mpi_communicator);
+    homogeneous_constraints.make_consistent_in_parallel(
+      locally_owned_dofs,
+      hanging_node_constraints.get_local_lines(),
+      mpi_communicator);
```

Comment:
This was bug 1: We want to merge from `hanging_node_constraints`, but due to the call `hanging_node_constraints.make_consistent_in_parallel()` a few lines up we might end up changing the local lines stored in that object. However, if that is the case, we cannot just merge the constraints. While one could try to set the flag `allow_different_local_lines` in the merge function, I am not sure how it would work, so I think it is much better to teach the readers to use the same representation of constraints if they choose to do so.